opengrok-web/src/test/java/org/opengrok/web/api/v1/controller/FileControllerTest.java

```diff
@@ -0,0 +1,93 @@
+package org.opengrok.web.api.v1.controller;
+
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.test.JerseyTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengrok.indexer.configuration.RuntimeEnvironment;
+import org.opengrok.indexer.history.HistoryGuru;
+import org.opengrok.indexer.history.RepositoryFactory;
+import org.opengrok.indexer.index.Indexer;
+import org.opengrok.indexer.util.TestRepository;
+import org.opengrok.web.api.v1.controller.FileController.LineDTO;
+
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.GenericType;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.junit.Assert.assertEquals;
+
+public class FileControllerTest extends JerseyTest {
```

Comment:
Any benefit to testing conditions that result in 400s or 404s?

Reply:
The problem is that Grizzly used by JerseyTest does not set neither `request` or `response`.